pkg/validation/validate.go

```diff
@@ -43,6 +43,16 @@ const (
 	DuplicateLabelNamesErrorMsg = "stream '%s' has duplicate label name: '%s'"
 )
 
+// TruncatedBytes is a metric of the total truncated bytes, by reason.
+var TruncatedBytes = prometheus.NewCounterVec(
```

Comment:
Might be worth adding a metric for number of lines truncated too; just knowing the bytes won't give any indication of how often this occurs (or at least will make it harder to determine)

Reply:
We already have `loki_discarded_{samples,bytes}_total`, which we partition by `user` and a `reason`. Perhaps we should start a `loki_mutated_{samples,bytes}_total` with a `reason=truncated` for consistency (it also leaves open the possibility of adding new things into this model in the future).

Reply:
Sounds reasonable to me :+1: